packages/puppeteer-core/src/common/ExecutionContext.ts

```diff
@@ -92,16 +94,35 @@ export class ExecutionContext {
 
   #puppeteerUtil?: Promise<JSHandle<PuppeteerUtil>>;
   get puppeteerUtil(): Promise<JSHandle<PuppeteerUtil>> {
-    if (!this.#puppeteerUtil) {
-      this.#puppeteerUtil = this.evaluateHandle(
-        `(() => {
-            const module = {};
-            ${injectedSource}
-            return module.exports.default;
-          })()`
-      ) as Promise<JSHandle<PuppeteerUtil>>;
+    scriptInjector.inject(async script => {
+      if (this.#puppeteerUtil) {
```

Comment:
looks like potential rejections here would be unhandled promise rejections. let's make sure all promises are caught here.